assets/app/features/upgrade/controllers/upgrade7-openstack-services.controller.spec.js

```diff
@@ -0,0 +1,263 @@
+                assert.isFalse(controller.openStackServices.checks.backup.status);                         
+            });
+
+        });
+
+        describe('when services check pass successfull and backup service call fails', function () {
+            beforeEach(function () {
+                bard.mockService(openStackFactory, {
+                    stopOpenstackServices: $q.when(passingOpenStackServicesResponse),
+                    createOpenstackBackup: $q.reject(failingResponse)
+                });
+                controller.openStackServices.stopOpenStackServices();
+                $rootScope.$digest();
+            });
+
+            it('should set openStackServices.completed status to true', function () {
+                assert.isTrue(controller.openStackServices.completed);
+            });
+
+            it('should update valid attribute of checks model to false', function () {
+                assert.isFalse(controller.openStackServices.valid);
+            });
+
+            it('should set running to false', function () {
+                assert.isFalse(controller.openStackServices.running);
+            });
+
+            it('should update services check value to true', function () {
+                assert.isTrue(controller.openStackServices.checks.services.status);   
+            });
+
+            it('should call backup service', function () {
+                assert.isTrue(openStackFactory.createOpenstackBackup.calledOnce);                         
+            });
+
+            it('should expose the errors through vm.openStackServices.errors object', function () {
+                expect(controller.openStackServices.errors).toEqual(failingResponse.data.errors);
+            });
+
+        });
+
+        describe('when services checks fails', function () {
+            beforeEach(function () {
+                bard.mockService(openStackFactory, {
+                    stopOpenstackServices: $q.when(failingOpenStackServicesResponse)
+                });
+                controller.openStackServices.stopOpenStackServices();
+                $rootScope.$digest();
+            });
+
+            it('should set openStackServices.completed status to true', function () {
+                assert.isTrue(controller.openStackServices.completed);
+            });
+
+            it('should update valid attribute of checks model to false', function () {
+                assert.isFalse(controller.openStackServices.valid);
+            });
+
+            it('should set running to false', function () {
+                assert.isFalse(controller.openStackServices.running);
+            });
+
+            it('should update checks values to false', function () {
+                assert.isFalse(controller.openStackServices.checks.services.status);
+            });
+
+            it('should not call backup service', function () {
+                assert.isFalse(openStackFactory.createOpenstackBackup.calledOnce);                         
```

Comment:
I think this is what you're trying to do: `expect(openStackFactory.createOpenstackBackup).not.toHaveBeenCalled();`